### PR TITLE
chore: upgrade golang 1.20.5 -> 1.20.10 (2.x)

### PR DESCRIPTION
**Pull request by bnpfeife**

### CODE REVIEW

**Review by alespour — APPROVED** [review]
LGTM, just curious, why not use `go1.20.10-latest` tag instead?

**Comment by alespour** [comment]
Closes https://github.com/influxdata/edge/issues/515

File: go.mod [hunk 1]
```diff
@@ -208,10 +208,10 @@ require (
 	golang.org/x/exp v0.0.0-20211216164055-b2b84827b756 // indirect
 	golang.org/x/exp/typeparams v0.0.0-20221208152030-732eee02a75a // indirect
 	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
-	golang.org/x/mod v0.7.0 // indirect
-	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/mod v0.13.0 // indirect
+	golang.org/x/net v0.16.0 // indirect
```
Comment (jdstrand) [inline go.mod:212]: Can we get this updated to v0.17.0? See https://github.com/advisories?query=GHSA-4374-p667-p6c8

Comment (jdstrand) [reply go.mod:212]:
```
$ go mod why -m golang.org/x/net
# golang.org/x/net
github.com/influxdata/influxdb/v2/vault
github.com/hashicorp/vault/api
golang.org/x/net/http2
```

Looks like hashicorp hasn't updated: https://github.com/hashicorp/vault/blob/main/go.mod. I suggest:
```
$ go mod edit -require golang.org/x/net@v0.17.0
$ go mod tidy
```

Which results in:
```
$ git diff
diff --git a/go.mod b/go.mod
index 679f076543..1bb944c0cc 100644
--- a/go.mod
+++ b/go.mod
@@ -209,7 +209,7 @@ require (
        golang.org/x/exp/typeparams v0.0.0-20221208152030-732eee02a75a // indirect
        golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
        golang.org/x/mod v0.13.0 // indirect
-       golang.org/x/net v0.16.0 // indirect
+       golang.org/x/net v0.17.0 // indirect
        golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f // indirect
        golang.org/x/term v0.13.0 // indirect
        golang.org/x/xerrors v0.0.0-20220411194840-2f41105eb62f // indirect
diff --git a/go.sum b/go.sum
index ab6488ec26..eff4774123 100644
--- a/go.sum
+++ b/go.sum
@@ -1099,8 +1099,8 @@ golang.org/x/net v0.0.0-20211015210444-4f30a5c0130f/go.mod h1:9nx3DQGgdP8bBQD5qx
 golang.org/x/net v0.0.0-20211112202133-69e39bad7dc2/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
 golang.org/x/net v0.0.0-20211118161319-6a13c67c3ce4/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
 golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd/go.mod h1:CfG3xpIq0wQ8r1q4Su4UZFWDARRcnwPjda9FqA0JpMk=
-golang.org/x/net v0.16.0 h1:7eBu7KsSvFDtSXUIDbh3aqlK4DPsZ1rByC8PFfBThos=
-golang.org/x/net v0.16.0/go.mod h1:NxSsAGuq816PNPmqtQdLE42eU2Fs7NoRIZrHJAlaCOE=
+golang.org/x/net v0.17.0 h1:pVaXccu2ozPjCXewfr1S7xza/zcXTity9cCdXQYSjIM=
+golang.org/x/net v0.17.0/go.mod h1:NxSsAGuq816PNPmqtQdLE42eU2Fs7NoRIZrHJAlaCOE=
 golang.org/x/oauth2 v0.0.0-20180821212333-d2e6202438be/go.mod h1:N/0e6XlmueqKjAGxoOufVs8QHGRruUQn6yWY3a++T0U=
 golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
 golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45/go.mod h1:gOpvHmFTYa4IltrdGE7lF6nIHvwfUNPOp7c8zoXwtLw=
```

Comment (bnpfeife) [reply go.mod]: All right, I've upgraded `golang.org/x/net` to `v0.17.0` in all of the PRs!

**Pull request merged**